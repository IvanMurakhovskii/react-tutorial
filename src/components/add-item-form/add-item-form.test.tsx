/**
 * @jest-environment jsdom
*/

import React from 'react';
import { mount } from "enzyme";
import AddItemForm from './add-item-form';

const mockMyEventHandler = jest.fn();

const props = {
    addItem: mockMyEventHandler
}

const item = <AddItemForm {...props} />

const wrapper = mount(item);

describe("AddItemForm", () => {
    it("render item with data", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('calls onSubmit function have been called once', () => {
        wrapper.find(AddItemForm).simulate('submit');
        expect(mockMyEventHandler).toHaveBeenCalledTimes(1);
    });

});