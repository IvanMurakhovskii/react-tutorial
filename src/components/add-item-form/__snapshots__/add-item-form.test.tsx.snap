// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddItemForm render item with data 1`] = `ReactWrapper {}`;
